src/org/mirah/jvm/compiler/method_state.mirah

```diff
@@ -48,13 +50,15 @@ class MethodState
   def conflictsWith(other:MethodState):Kind
     return nil unless @name.equals(other.name)
     return nil unless @num_args == other.num_args
-    if @signature.nil? && other.signature
+    if self.isMacro && other.signature
       other.conflictsWith(self)
-    elsif @signature.nil? || other.signature.nil?
+    elsif self.isMacro || other.isMacro
       if @num_args == 0
-        # We know there's a conflict
-        # TODO: unless these are macros and one of them is static
-        Kind.ERROR
+        if self.isMacro && other.isMacro && (self.static ^ other.static) # Macros are allowed to have overlapping no-arg signatures if exactgly one of them is static.
+          nil
+        else                                                             # we know there's a conflict
+          Kind.ERROR
+        end
```

Comment:
Oh! How about this:
```mirah
if self.isMacro && !other.isMacro
  other.conflictsWith self
elsif !self.isMacro && other.isMacro
  if @num_args != 0
    # At least one of these is a macro, so it's hard to tell if
    # the arguments will actually conflict. Just emit a warning.
    Kind.WARNING
  else
    Kind.ERROR
  end
elsif self.isMacro && other.isMacro
  if @num_args == 0
    if self.static ^ other.static 
      # Macros are allowed to have overlapping no-arg signatures if exactly one of them is static.
      nil
    else
      Kind.ERROR
    end
  else
    # Since both of these are macros, it's hard to tell if their arguments will conflict.
    Kind.WARNING
  end
else
# both are methods
...
end 
```